inc/namespace.php

```diff
@@ -456,3 +459,92 @@ function on_request_stats( TransferStats $stats ) {
 
 	on_aws_guzzle_request_stats( $stats );
 }
+
+/**
+ * Get the EC2 Instance metadata.
+ *
+ * This will cause a remote request to the metadata service when the cache is empty.
+ *
+ * @return array $array(
+ *    accountId: string,
+ *    architecture: string,
+ *    availabilityZone: string,
+ *    billingProducts: string,
+ *    devpayProductCodes: string,
+ *    marketplaceProductCodes: string,
+ *    imageId: string,
+ *    instanceId: string,
+ *    instanceType: string,
+ *    kernelId: string,
+ *    pendingTime: string,
+ *    privateIp: string,
+ *    ramdiskId: string,
+ *    region: string,
+ *    version: string,
+ * )
+ */
+function get_ec2_instance_metadata() : array {
+	$has_cache = false;
+	$cache_key = 'altis.ec2_instance_metadata';
+	$cached_data = apcu_fetch( $cache_key, $has_cache );
```

Comment:
For my own understanding why do we need to use `apcu` for this?

Also it looks like the `apcu` package was only just added - should we add a `function_exists()` check here?

Reply:
This is so it's specifically a per-machine cache I think, whereas the object cache is per-stack.

We should already have had apcu installed I think? We're using it on humanmade-production-v2, so it's been around for a while afaik.

Reply:
Yeah apcu has been there for some time, we just didn't have it documented. It is indeed to cache per-machine. I'll add an inline comment.